src/index.ts

```diff
@@ -135,14 +137,15 @@ export function resolve(
   found: boolean
   path?: string | null
 } {
-  if (!cachedOptions || cachedOptions !== options) {
+  if (!cachedOptions || previousOptions !== options) {
+    previousOptions = options
     cachedOptions = {
       ...options,
       conditionNames: options?.conditionNames ?? defaultConditionNames,
       extensions: options?.extensions ?? defaultExtensions,
       extensionAlias: options?.extensionAlias ?? defaultExtensionAlias,
       mainFields: options?.mainFields ?? defaultMainFields,
-      fileSystem,
+      fileSystem: new CachedInputFileSystem(fileSystem, 4000),
```

Comment:
I just copied the 4000 value from the default instructions. I'm not sure how much this will affect in practice but it does save time.